core/services/workflows/syncer/workflow_registry.go

```diff
@@ -499,6 +499,8 @@ func (w *workflowRegistry) workflowMetadataToEvents(ctx context.Context, workflo
 					Data:      toRegisteredEvent,
 					EventType: WorkflowRegisteredEvent,
 				})
+			case nil:
```

Comment:
You're handling case nil and default the same, which implies that a nil error and an error that isn't errNotFound (also not the lint error) won't get handled the same

That doesn't seem right and also doesn't reflect what the comment says